zrml/swaps/src/lib.rs

```diff
@@ -1434,7 +1440,17 @@ mod pallet {
             }
 
             if *asset_in == base_asset {
-                T::RikiddoSigmoidFeeMarketEma::price(*pool_id, balance_out, &balances)
+                let price_with_fee =
+                    T::RikiddoSigmoidFeeMarketEma::price(*pool_id, balance_out, &balances)?;
+                if with_fees {
+                    Ok(price_with_fee.saturated_into())
+                } else {
+                    let fee_pct = T::RikiddoSigmoidFeeMarketEma::fee(*pool_id)?.saturated_into();
+                    let fee_plus_one = BASE.saturating_add(fee_pct);
+                    let price_without_fee: u128 =
+                        bdiv(price_with_fee.saturated_into(), fee_plus_one)?;
+                    Ok(price_without_fee.saturated_into())
+                }
```

Comment:
I think all these changes regarding Rikiddo should be reverted. It doesn't really matter what the Rikiddo behavior is right now, and it seems to me like this calculation is not correct.

@sea212 It seems like the term _fee_ is used as a name for `min(min_revenue, initial_fee, eta(r))`, where `eta` is the sigmoid function (ignoring the fact that `r` is not always well-defined), so if you multiply this with the total issuance of all outcome tokens, you get the exponent in the cost/price formula. It seems like it is used here as something like an approximation of the actual "fee" of Rikiddo. This seems off.

I'm fairly sure that one shouldn't even speak of a "fee" when dealing with LMSR and its variants. For example, in terms of LS-LMSR, if you buy from the market and then immediately sell back to the market, you make no profit or loss. This wouldn't be the case if there are fees. On low volume trades, the same should be true for Rikiddo. Have you or @numacoding ever defined an actual fee for Rikiddo in abstract terms?

Reply:
The first usage of the term "fee" was defined in the [Rikiddo research report from Numa](https://docs.google.com/document/d/1GF29lC2t987Knk57RZPRLOjNTdI2G-q9gs3_nufRCss/edit).

It is actually `min(min_revenue, initial_fee + eta(r))`. `min_revenue` is what Numa calls "minimal revenue coefficient (omega)".

It is some kind of approximation. This part definitely has to be improved, but for now we can leave it like that and raise an issue to handle that imo. (since Rikiddo is not active).
I think in a path-independent algorithm like LS-LMSR the assumption that there is no profit or loss is correct, because the amount of base currency to pay for buying `C(q0) -> C(q1)` is `C(q1) - C(q0)` and the amount received for selling `C(q1) -> C(q0)` is `C(q1) - C(q0)`. However, with Rikiddo this does not hold, because the "fee" parameter can vary in between two trades and change `C(q)`. It depends on additional data, I think it would be cleaner if this would be indicated in the function, like `C(q, r)`.
Although the question was raised what kind of implications the loss of the path-independence property has, it was never properly examined.

Reply:
Thanks for the answer. Can you provide access to the research report. I seem to be locked out. :(

Indeed, we should start writing `C(q, r)` or `C_r(q)` (note: `C_r`, with fixed `r` is actually path-independent). As you say, for path-independent buying and then selling results in an exact break-even (because of, well, path-independence). Which means that there are no fees involved.

My issue here is that if the buy _and_ the sell price go up, then whenever you sell to the market, the market creators pay _you_ a fee. That just feels wrong. I think that 3.3.1 of _A Practical Liquidity-Sensitive Automated Market Maker_ provides a somewhat servicable solution to this problem.

Reply:
@vivekvpandya so maybe a TODO with an issue number would be appropriate.